applications/vessel-history/src/service-worker.ts

```diff
@@ -0,0 +1,163 @@
+// are fulfilled with your index.html shell. Learn more at
+// https://developers.google.com/web/fundamentals/architecture/app-shell
+const fileExtensionRegexp = new RegExp('/[^/?]+\\.[^/]+$')
+registerRoute(
+  // Return false to exempt requests from being fulfilled by index.html.
+  ({ request, url }: { request: Request; url: URL }) => {
+    // If this isn't a navigation, skip.
+    if (request.mode !== 'navigate') {
+      return false
+    }
+
+    // If this is a URL that starts with /_, skip.
+    if (url.pathname.startsWith('/_')) {
+      return false
+    }
+
+    // If this looks like a URL for a resource, because it contains
+    // a file extension, skip.
+    if (url.pathname.match(fileExtensionRegexp)) {
+      return false
+    }
+
+    // Return true to signal that we want to use the handler.
+    return true
+  },
+  createHandlerBoundToURL(process.env.PUBLIC_URL + '/index.html')
+)
+
+// This allows the web app to trigger skipWaiting via
+// registration.waiting.postMessage({type: 'SKIP_WAITING'})
+self.addEventListener('message', (event) => {
+  if (event.data && event.data.type === 'SKIP_WAITING') {
+    self.skipWaiting()
+  }
+})
+
+// Any other custom service worker logic can go here.
+
+// Cache all requests to /public folder like /icons and manifest.jsonm
+registerRoute(
+  ({ url }) =>
+    (url.origin === self.location.origin && url.pathname.startsWith('/icons')) ||
+    url.pathname.endsWith('manifest.json'),
+  new StaleWhileRevalidate({
+    cacheName: 'public',
+    plugins: [
+      // Ensure that once this runtime cache reaches a maximum size the
+      // least-recently used images are removed.
+      new ExpirationPlugin({ maxEntries: 100 }),
+      new BackgroundSyncPlugin('public-bg-sync', {
+        maxRetentionTime: 24 * 60, // Retry for max of 24 Hours (specified in minutes)
+      }),
+    ],
+  })
+)
+
+// Cache gstatic resources (css and fonts)
+registerRoute(
+  ({ url }) => ['https://fonts.googleapis.com', 'https://fonts.gstatic.com'].includes(url.origin),
+  new StaleWhileRevalidate({
+    cacheName: 'fonts',
+    plugins: [
+      // Ensure that once this runtime cache reaches a maximum size the
+      // least-recently used fonts are removed.
+      new ExpirationPlugin({ maxEntries: 5 }),
+      new BackgroundSyncPlugin('fonts-bg-sync', {
+        maxRetentionTime: 24 * 60, // Retry for max of 24 Hours (specified in minutes)
+      }),
+    ],
+  })
+)
+
+// Cache locales
+registerRoute(
+  /\/locales\/.*/,
+  new StaleWhileRevalidate({
+    cacheName: 'locales',
+    plugins: [
+      new ExpirationPlugin({ maxEntries: 50 }),
+      new BackgroundSyncPlugin('locales-bg-sync', {
+        maxRetentionTime: 24 * 60, // Retry for max of 24 Hours (specified in minutes)
+      }),
+    ],
+  })
+)
+
+// Cache API Auth Calls
+registerRoute(
+  ({ url }) =>
+    process.env.REACT_APP_API_GATEWAY &&
+    url.origin === process.env.REACT_APP_API_GATEWAY &&
```

Comment:
to avoid duplicating this in the auth and the regular API calls what about having a global const like
```const API_GATEWAY = process.env.REACT_APP_API_GATEWAY```
and we check it here and [here](https://github.com/GlobalFishingWatch/frontend/pull/421/files#diff-d2f15e6637529cd54a564a392ebcf7aac0d10c85c4266eefbcb0017d2e1972f7R138) it?